clients/admin-ui/src/pages/subject-request/[id].tsx

```diff
@@ -8,94 +8,67 @@ import {
 } from '@fidesui/react';
 import type { NextPage } from 'next';
 import Head from 'next/head';
-import { getSession } from 'next-auth/react';
+import { useRouter } from 'next/router';
 import React from 'react';
 
-import { useAppSelector } from '../../app/hooks';
-import { wrapper } from '../../app/store';
+import ProtectedRoute from '../../features/auth/ProtectedRoute';
 import NavBar from '../../features/common/NavBar';
-import {
-  privacyRequestApi,
-  selectPrivacyRequestFilters,
-  setRequestId,
-  setVerbose,
-  useGetAllPrivacyRequestsQuery,
-} from '../../features/privacy-requests';
+import { useGetAllPrivacyRequestsQuery } from '../../features/privacy-requests';
 import SubjectRequest from '../../features/subject-request/SubjectRequest';
-import { assignToken, setUser } from '../../features/user';
 
-const SubjectRequestDetails: NextPage<{}> = () => {
-  const filters = useAppSelector(selectPrivacyRequestFilters);
-  const { data } = useGetAllPrivacyRequestsQuery(filters);
+const useSubjectRequestDetails = () => {
+  const router = useRouter();
+  const { id = '' } = router.query;
+  const { data } = useGetAllPrivacyRequestsQuery({
+    id: Array.isArray(id) ? id[0] : id,
+    verbose: true,
+  });
+
+  return data;
+};
+
+const SubjectRequestDetails: NextPage = () => {
+  const data = useSubjectRequestDetails();
   const body =
-    data?.items.length === 0 ? (
+    !data || data?.items.length === 0 ? (
       <Text>404 no subject request found</Text>
```

Comment:
It looks like this page is flashing this message for a split second while the page is loading the data. If the user has a slow connection it will display the 404 message while the page is loading. I guess we'll need to incorporate some kind of loading behavior now since we're no longer doing SSR. That might be out of scope of this PR though

Reply:
Good spot. Between Chakra UI and Redux Toolkit Query we actually have everything we need to solve this in like 5min, so I pushed up a quick fix for this. :)